pages/activityList.js

```diff
@@ -35,26 +36,127 @@ const StyledLi = styled.li`
+  display: block;
+  text-align: center;
+  font-size: medium;
+  font-weight: bold;
+`;
+
+const WrapperSearchBar = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const InputSearchField = styled.input`
+  padding: 8px 8px 8px 40px;
+  border: 2px solid black;
+  border-radius: 0.5rem;
+  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath d='M23.707,22.293l-5.969-5.969a10.016,10.016,0,1,0-1.414,1.414l5.969,5.969a1,1,0,0,0,1.414-1.414ZM10,18a8,8,0,1,1,8-8A8.009,8.009,0,0,1,10,18Z' fill='black'/%3E%3C/svg%3E");
+  background-repeat: no-repeat;
+  background-position: 10px center;
+  background-size: 20px;
+  width: 310px;
+  margin-bottom: 25px;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.1);
+`;
+
+const ErrorMessage = styled.p`
+  color: red;
+  font-size: medium;
+  font-weight: bold;
+  text-align: center;
+  margin: 0 0 25px 0;
+`;
+
 export default function ActivityList({ activities }) {
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleOnChange = (event) => {
+    setSearchTerm(event.target.value);
+  };
+
+  const getFilteredActivities = (searchTerm, activities) => {
+    if (!searchTerm) {
+      return activities;
+    }
+
+    return activities.filter(
+      (activity) =>
+        activity.category.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        activity.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        activity.area.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        activity.country.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+  };
+
+  const filteredActivities = getFilteredActivities(searchTerm, activities);
+
   return (
     <>
       <StyledHeadline>List of Activities</StyledHeadline>
-      <StyledUl>
-        {activities.map((activity) => (
-          <>
-            <StyledLinkDetailsPage href={`/${activity.id}`}>
-              <StyledLi key={activity.id}>
-                <ActivityCard
-                  id={activity.id}
-                  image={activity.image}
-                  title={activity.title}
-                  area={activity.area}
-                />
-              </StyledLi>
-            </StyledLinkDetailsPage>
-          </>
-        ))}
-      </StyledUl>
+      <StyledSection>
+        <LabelSearchField htmlFor="image">Filter Activities</LabelSearchField>
+        <WrapperSearchBar>
+          <InputSearchField
+            id="image"
+            name="image"
+            type="search"
+            value={searchTerm}
+            onChange={handleOnChange}
+            placeholder="Type to filter activities"
+            required
+          />
+        </WrapperSearchBar>
+      </StyledSection>
+      {filteredActivities && filteredActivities.length > 0 ? (
+        <StyledUl>
+          {filteredActivities.map((activity) => (
+            <>
+              <StyledLinkDetailsPage href={`/${activity.id}`}>
```

Comment:
For valid HTML <li> tags always have to be the direct child of an <ul>/<ol>/<menu> tag.
At the moment your list looks like this:

<ul>
  <a>
     <li></li>
  </a>
</ul>
But it has to be like this:

<ul>
  <li>
     <a></a>
  </li>
</ul>